### PR TITLE
Use new access levels permissions

### PR DESCRIPTION
**Pull request by mcanevet**
https://github.com/xanzy/go-gitlab/blob/cd1e3f03a766cd7f9a7d2257926039b77f2c7710/gitlab.go#L66-L77

### CODE REVIEW

**Comment by roidelapluie** [comment]
Can  you please tell me which release of gitlab is required to support MAINTAINER? for the changelog.

**Comment by roidelapluie** [comment]
Also if by accident current gitlab still allows master in the API, let's keep master

File: gitlab/util.go [hunk 1]
```diff
@@ -11,19 +11,24 @@ import (
 )
 
 var accessLevelNameToValue = map[string]gitlab.AccessLevelValue{
-	"guest":     gitlab.GuestPermissions,
-	"reporter":  gitlab.ReporterPermissions,
-	"developer": gitlab.DeveloperPermissions,
-	"master":    gitlab.MasterPermissions,
```
Comment (pdecat) [inline gitlab/util.go]: As @roidelapluie commented, this should probably be kept for backward compatibility.

File: gitlab/util.go [hunk 2]
```diff
@@ -11,19 +11,24 @@ import (
...
-	"owner":     gitlab.OwnerPermission,
+	"no one":     gitlab.NoPermissions,
+	"guest":      gitlab.GuestPermissions,
+	"reporter":   gitlab.ReporterPermissions,
+	"developer":  gitlab.DeveloperPermissions,
+	"maintainer": gitlab.MaintainerPermissions,
```
Comment (pdecat) [inline gitlab/util.go]:
```suggestion
 	"master":    gitlab.MaintainerPermissions,
	"maintainer": gitlab.MaintainerPermissions,
```

File: gitlab/util.go [hunk 3]
```diff
@@ -106,17 +111,22 @@ func buildTwoPartID(a, b *string) string {
...
```
Comment (pdecat) [inline gitlab/util.go]:
```suggestion
 	"master":     gitlab.MaintainerPermissions,
 	"maintainer": gitlab.MaintainerPermissions,
```

File: gitlab/util.go [hunk 4]
```diff
@@ -106,17 +111,22 @@ func buildTwoPartID(a, b *string) string {
...
+	"owner":      gitlab.OwnerPermission,
+
+	// Deprecated
+	"master": gitlab.MasterPermissions,
```
Comment (pdecat) [inline gitlab/util.go]: Oops, sorry! :confused:

**Comment by mcanevet** [comment]
@roidelapluie renaming of master role to maintainer occurred in v11.0 : https://gitlab.com/gitlab-org/gitlab-ce/merge_requests/19080 but this should be backward compatible as internally it uses the same value (40).

File: gitlab/util.go [hunk 5]
```diff
@@ -11,19 +11,24 @@ import (
...
-	"owner":     gitlab.OwnerPermission,
+	"no one":     gitlab.NoPermissions,
+	"guest":      gitlab.GuestPermissions,
+	"reporter":   gitlab.ReporterPermissions,
+	"developer":  gitlab.DeveloperPermissions,
+	"maintainer": gitlab.MaintainerPermissions,
+	"owner":      gitlab.OwnerPermission,
+
+	// Deprecated
+	"master": gitlab.MaintainerPermissions,
```
Comment (roidelapluie) [inline gitlab/util.go]: I like this!!

**Pull request merged**